File: C#Part2 - Homework/01. Arrays - Homework/04.MaximalSequence/MaximalSequence.cs
namespace MaximalSequence
{
    using System;

    class MaximalSequence
    {
        // Write a program that finds the maximal sequence of equal elements in an array.

        static void Main()
        {
            Console.Write("Enter size of the array =");
            int n = int.Parse(Console.ReadLine());
            int[] array = new int[n];

            int currentSequence = 1;
            int bestSequence = 1;
            int numberAIndex = 0;

            for (int i = 0; i < array.Length; i++)
            {
                array[i] = int.Parse(Console.ReadLine());
            }


            for (int i = 0; i < n - 1; i++)
            {
                if (array[i] == array[i + 1])
                {
                    currentSequence++;
                }
                else
                {
                    currentSequence = 1;
                }
                if (currentSequence >= bestSequence)
                {
                    bestSequence = currentSequence;
                    numberAIndex = array[i];
                }
            }

            for (int j = 0; j < bestSequence; j++)
            {
                Console.Write("{0}", numberAIndex);
            }
            Console.WriteLine();
        }
    }
}

File: JavaScript/Homework/Arrays - Homework/Problem 3 Maximal sequence/MaximalSequence.js
// Write a script that finds the maximal sequence of equal elements in an array.

var arr,
    arrSequence,
    i,
    len,
    count =1,
    bestSquence = 1,
    num = 0;

arr = [2, 3, 5, 6, 2, 2, 2, 2, 1];
arrSequence = [];

for (i = 1, len = arr.length-1; i <= len; i+=1) {
    if (arr[i] === arr[i-1]) {
        count += 1;
    } else {
        count = 1;
    }
    if(count >= bestSquence){
        bestSquence = count;
        num = arr[i];
    }
}

for (i = 0; i < bestSquence; i+=1) {
    arrSequence[i] = num;
}
console.log(arrSequence);